app/controllers/home_pages_controller.rb

```diff
@@ -0,0 +1,33 @@
+class HomePagesController < ApplicationController
+  def index
+    @home_pages = current_user.home_pages.order(created_at: :desc).all
+    @total_amount = []
+
+    @home_pages.each do |category|
+      total = 0
+      category.expenses_categories.each do |expense_category|
+        total += expense_category.expense.amount
+      end
+      @total_amount.push(total)
+    end
```

Comment:
[Optional]

- [ ] It will be better if you can make these operations in your models instead of controllers. Having data-related operations/methods in our controller will make our applications prone to error and hard to test. You can simply create a method in your model that does the same thing and call that method here.